src/main.cpp

```diff
@@ -2940,11 +2943,28 @@ bool Session::SeekTime(double seekTime, unsigned int streamId, bool preceeding)
     }
   }
 
+  // correct for starting segment pts value of chapter and chapter offset within program
   uint64_t seekTimeCorrected = static_cast<uint64_t>(seekTime * STREAM_TIME_BASE);
+  int64_t ptsDiff = 0;
   if (timing_stream_)
   {
+    // after seeking across chapters with fmp4 streams the reader will not have started
+    // so we start here to ensure that we have the required information to correctly
+    // seek with proper stream alignment
+    if (!timing_stream_->reader_->IsStarted())
+    {
+      bool bReset;
+      timing_stream_->stream_.seek_time(
+          static_cast<double>(seekTimeCorrected + timing_stream_->stream_.GetAbsolutePTSOffset()) /
+              STREAM_TIME_BASE,
+          preceeding, bReset);
+      if (bReset)
+        timing_stream_->reader_->Reset(false);
+      bool bStarted;
+      timing_stream_->reader_->Start(bStarted);
```

Comment:
This makes no sense and is bad code. bStarted could have any value as it's a basic type. It should be initialised. Plus it's copied in the call to start so is not used at all!